pkg/controlplane/client.go

```diff
@@ -113,11 +114,23 @@ func (c *client) IsActive() bool {
 
 // setActive the peer status (active or not).
 func (c *client) setActive(active bool) {
-	c.lock.Lock()
-	defer c.lock.Unlock()
-	c.active = active
-	if active || c.lastSeen.IsZero() {
-		c.lastSeen = time.Now()
+	var activePrevState bool
+	{
+		c.lock.Lock()
+		defer c.lock.Unlock()
+		activePrevState = c.active
+		c.active = active
+		if active || c.lastSeen.IsZero() {
+			c.lastSeen = time.Now()
+		}
+	}
+	// Update other components like the policy engine with the peer status.
+	if !activePrevState && active {
```

Comment:
```suggestion
	if active != activePrevState {
	   c.c.peerStatusCallback(active)
	}
```
can be an alternative to L128-134